### PR TITLE
[Web] Stylus support 

### PR DESCRIPTION
**Pull request by m-bert**
## Description

This PR adds more information about `stylus` to event. `Pan` and `Hover` events now have `stylusData` field, which contains the following information:

- `tiltX`
- `tiltY`
- `altitudeAngle`
- `azimuthAngle`
- `pressure`

>[!IMPORTANT]
> Because we receive only one set of data (either `tiltX/tiltY` or `altitudeAngle/azimuthAngle`, we use [this conversion algorithm](https://w3c.github.io/pointerevents/#converting-between-tiltx-tilty-and-altitudeangle-azimuthangle) to calculate second set.

>[!NOTE]
> This PR adds `stylusData` only on web, native platforms will be handled in another PRs
## Test plan

Tested on newly added example.

### CODE REVIEW

**Review by latekvo — APPROVED** [review]
All good 👍

File: src/web/tools/PointerEventManager.ts [hunk 1]
```diff
@@ -88,13 +85,6 @@ export default class PointerEventManager extends EventManager<HTMLElement> {
     // Stylus triggers `pointermove` event when it detects changes in pressure. Since it is very sensitive to those changes,
     // it constantly sends events, even though there was no change in position. To fix that we check whether
     // pointer has actually moved and if not, we do not send event.
```
Comment (j-piasecki) [inline src/web/tools/PointerEventManager.ts:87]: Not true anymore

File: src/web/utils.ts [hunk 2]
```diff
@@ -54,3 +54,177 @@ export function calculateViewScale(view: HTMLElement) {
...
+
+  if (event.tiltX === 0 && event.tiltY === 0) {
+    // If we are in this branch, it means that either tilt properties are not supported and we have to calculate them from altitude and azimuth angles,
+    // or stylus is perpendicular to the screen and we can use altitude / azimuth instead of tilt
+
+    // If azimuth and altitude are undefined in this branch, it means that we are either perpendicular to the screen,
+    // or that none of the position sets is supported. In that case, we can treat stylus as perpendicular
+    //
+    // @ts-ignore This property exists (https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent#instance_properties)
+    if (event.azimuthAngle === undefined || event.altitudeAngle === undefined) {
```
Comment (j-piasecki) [inline src/web/utils.ts:75]: You can extract those to variables instead of `@ts-ignoring` multiple times

Comment (m-bert) [reply src/web/utils.ts]: Good idea. Done in [9fa4003](https://github.com/software-mansion/react-native-gesture-handler/pull/3107/commits/9fa4003dae72bdad1de3d9323fe6dcac43562963)

File: src/web/utils.ts [hunk 3]
```diff
@@ -64,15 +64,18 @@ export function tryExtractStylusData(
     return;
   }
 
+  // @ts-ignore This property exists (https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent#instance_properties)
+  const eventAzimuthAngle = event.azimuthAngle;
+  // @ts-ignore This property exists (https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent#instance_properties)
+  const eventAltitudeAngle = event.altitudeAngle;
```
Comment (j-piasecki) [inline src/web/utils.ts:70]: I think those will be typed as `any`, if so explicitly setting them to `number | undefined` may be a good idea.

**Review by j-piasecki — APPROVED** [review]

File: src/web/utils.ts (hunk 3 above)
Comment (m-bert) [reply src/web/utils.ts]: I agree, done in [18d137f](https://github.com/software-mansion/react-native-gesture-handler/pull/3107/commits/18d137f8fa9cb0e9e3c9e6edace10f314dceb12d)

**Pull request merged**